slither/tools/read_storage/read_storage.py

```diff
@@ -111,7 +119,35 @@ def get_storage_layout(self) -> None:
                 elif isinstance(type_, ArrayType):
                     elems = self._all_array_slots(var, contract, type_, info.slot)
                     tmp[var.name].elems = elems
-
+        for contract, var in self.constant_slots:
+            var_name = var.name
+            try:
+                exp = var.expression
+                if isinstance(exp, CallExpression) and str(exp.called) == "keccak256(bytes)":
+                    exp = exp.arguments[0]
+                if isinstance(exp, Literal):
+                    if exp.value.startswith("0x"):
+                        slot = coerce_type("int", exp.value)
+                    else:
+                        slot_str = exp.value
+                        slot_hash = keccak(text=slot_str)
+                        slot = coerce_type("int", slot_hash)
+                else:
+                    continue
+                offset = 0
+                type_string, size = self.find_constant_slot_storage_type(var)
+                if type_string:
+                    tmp[var.name] = SlotInfo(
+                        name=var_name, type_string=type_string, slot=slot, size=size, offset=offset
+                    )
+                    self.log += (
+                        f"\nSlot Name: {var_name}\nType: bytes32"
+                        f"\nStorage Type: {type_string}\nSlot: {str(exp)}\n"
+                    )
+                    logger.info(self.log)
+                    self.log = ""
+            except TypeError:
```

Comment:
When is this `TypeError` being hit? It seems like it could be removed by performing more validations of what type `exp.value` is